pydantic/type_adapter.py

```diff
@@ -233,100 +208,129 @@ def __init__(
         self._type = type
         self._config = config
         self._parent_depth = _parent_depth
+
         if module is None:
             f = sys._getframe(1)
             self._module_name = cast(str, f.f_globals.get('__name__', ''))
         else:
             self._module_name = module
 
-        self._core_schema: CoreSchema | None = None
-        self._validator: SchemaValidator | PluggableSchemaValidator | None = None
-        self._serializer: SchemaSerializer | None = None
+        self.core_schema: CoreSchema
+        self.validator: SchemaValidator | PluggableSchemaValidator
+        self.serializer: SchemaSerializer
+
+        localns: _namespace_utils.MappingNamespace = (
+            _typing_extra.parent_frame_namespace(parent_depth=self._parent_depth) or {}
+        )
+        globalns: _namespace_utils.GlobalsNamespace = sys._getframe(max(self._parent_depth - 1, 1)).f_globals.copy()
```

Comment:
There is also `sys._getframe` use above for `_module_name` thing handling. For that the "depth" is `1`, but I dont understand why. Could we just do the `sys._getframe(max(self._parent_depth - 1, 1))` once and do the `_module_name` things from that? Would be much more clear.